account_allowance_adjust_transaction_unit_test.go

```diff
@@ -0,0 +1,206 @@
+//go:build all || unit
+// +build all unit
+
+package hedera
+
+import (
+	"fmt"
+	"testing"
+	"time"
+
+	"github.com/stretchr/testify/require"
+)
+
+/*-
+ *
+ * Hedera Go SDK
+ *
+ * Copyright (C) 2020 - 2022 Hedera Hashgraph, LLC
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *      http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ *
+ */
+
+func TestUnitAccountAllowanceAdjustTransactionGet(t *testing.T) {
+	key, err := PrivateKeyGenerate()
+	key2, err := PrivateKeyGenerate()
+	require.NoError(t, err)
+
+	nodeAccountIDs := []AccountID{{Account: 10}}
+	transactionID := TransactionIDGenerate(AccountID{Account: 123})
+	tokenID := TokenID{Token: 3}
+	NftID := NftID{tokenID, 1}
+	tx, err := NewAccountAllowanceAdjustTransaction().
+		AddHbarAllowance(nodeAccountIDs[0], HbarFromTinybar(1)).
+		AddTokenAllowance(tokenID, nodeAccountIDs[0], 1).
+		AddTokenNftAllowance(NftID, nodeAccountIDs[0]).
+		SetTransactionID(transactionID).SetNodeAccountIDs(nodeAccountIDs).
+		SetMaxTransactionFee(HbarFromTinybar(100)).SetRegenerateTransactionID(true).
+		SetTransactionMemo("go sdk unit test").SetTransactionValidDuration(time.Second * 120).
+		SetMaxRetry(1).SetMaxBackoff(time.Second * 120).SetMinBackoff(time.Second * 1).
+		Freeze()
+	sign, err := key2.SignTransaction(&tx.Transaction)
+	require.NoError(t, err)
+	tx.AddSignature(key.PublicKey(), sign)
```

Comment:
<!-- Questions are appropriate if you have a potential concern but are not quite sure if it's relevant or not. Asking the author for clarification or investigation can lead to a quick resolution.  -->
**question:** ‏Hmm, we use the signature that is generated with `key2.SignTransaction()` but with the other key? Actually why do we need two keys, can't we simplify the code and use only one?

Reply:
We need it for coverage.